dawn.json

```diff
@@ -910,20 +913,12 @@
     "shader module": {
         "category": "object"
     },
-    "shader module builder": {
-        "category": "object",
-        "methods": [
-            {
-                "name": "get result",
-                "returns": "shader module"
-            },
-            {
-                "name": "set source",
-                "args": [
-                    {"name": "code size", "type": "uint32_t"},
-                    {"name": "code", "type": "uint32_t", "annotation": "const*", "length": "code size"}
-                ]
-            }
+    "shader module descriptor": {
+        "category": "structure",
+        "extensible": true,
+        "members": [
+            {"name": "code size", "type": "uint32_t"},
```

Comment:
There is no validation because in the C API it isn't possible to have guarantees about the size of an array given as a pointer. For WebGPU there won't be a `codeSize`, instead `code` will be a typed array or something like that, and Blink will be handle the translation from `code` to `code + codeSize`.

Reply:
Ahh, makes sense. I was thinking you could do a strlen or something, but yeah, kind of pointless.